src/bootstrap/bin/main.rs

```diff
@@ -9,10 +9,26 @@ use std::env;
 
 use bootstrap::{Build, Config, Subcommand, VERSION};
 
-fn main() {
+fn main() -> std::io::Result<()> {
     let args = env::args().skip(1).collect::<Vec<_>>();
     let config = Config::parse(&args);
 
+    let mut build_lock;
+    let _build_lock_guard;
+    if cfg!(any(unix, windows)) {
+        build_lock = fd_lock::RwLock::new(std::fs::File::create(config.out.join("lock"))?);
+        _build_lock_guard = match build_lock.try_write() {
+            Ok(lock) => lock,
+            err => {
+                println!("warning: build directory locked, waiting for lock");
+                drop(err);
```

Comment:
I think you can avoid this drop by using a wildcard:
```suggestion
            Err(_) => {
                println!("warning: build directory locked, waiting for lock");
```

Reply:
@jyn514 That was the previous thing I tried, but if I do that, Rust still thinks the borrow exists (even though the borrow is in the `Ok` variant, Rust doesn't understand that).

Reply:
Ok, fair enough